dev/benchmarks/complex_layout/lib/main.dart

```diff
@@ -36,13 +37,45 @@ class ComplexLayoutAppState extends State<ComplexLayoutApp> {
     });
   }
 
+  ScrollMode _scrollMode = ScrollMode.complex;
+  ScrollMode get scrollMode => _scrollMode;
+  set scrollMode(ScrollMode mode) {
+    setState(() {
+      _scrollMode = mode;
+    });
+  }
+
   void toggleAnimationSpeed() {
     setState(() {
       timeDilation = (timeDilation != 1.0) ? 1.0 : 5.0;
     });
   }
 }
 
+class TileScrollLayout extends StatelessWidget {
+  const TileScrollLayout({ Key key }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+    for (int i = 0; i < 200; i++) {
```

Comment:
We should just use ListView.builder, it's dramatically more efficient and may in fact be simpler (no for loop). 200 is a huge number of widget trees to build every frame. If you want to build them ahead of time then you shouldn't do it in the build function, you should do it at compile time using const widgets in a const list, or maybe asynchronously over several frames, or some such.